### PR TITLE
fix(core): fix fd-calendar accessibility

### PR DESCRIPTION
**Pull request by dimamarksman**
## Related Issue.
Closes  https://github.com/SAP/fundamental-ngx/issues/6225

## Description

This PR fixes `fd-calendar` accessibility issues listed in the https://github.com/SAP/fundamental-ngx/issues/6225.
The issue about
`Extra information such as “Grid”, “column 1 row 1”, “contains controls” is read out when date is accessed.`
is not fixed cause this metadata is the screen reader native behavior.

##### During Implementation
1. Visual Testing:
- [na] visual misalignments/updates
- [na] check Light/Dark/HCB/HCW themes
- [na] RTL/LTR - proper rendering and labeling
- [na] responsiveness(resize)
- [na] Content Density (Cozy/Compact/(Condensed))
- [na] States - hover/disabled/focused/active/on click/selected/selected hover/press state
- [na] Interaction/Animation - open/close, expand/collapse, add/remove, check/uncheck
- [x] Mouse vs. Keyboard support
- [na] Text Truncation
2. API and functional correctness
- [x] check for console logs (warnings, errors)
- [na] API boundary values
- [na] different combinations of components - free style
- [na] change the API values during testing
3. Documentation and Example validations
- [x] missing API documentation or it is not understandable
- [na] poor examples
- [na] Stackblitz works for all examples
4. Accessibility testing
5. Browser Testing - Edge, Safari, Chrome, Firefox

##### PR Quality
- [x] the commit message(s) follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
- [x] Run npm run build-pack-library and test in external application
- [na] update `README.md`
- [na] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)

##### PR Review
- [na] visual misalignments/updates
- [na] check Light/Dark/HCB/HCW themes
- [na] RTL/LTR - proper rendering and labeling
- [na] responsiveness(resize)
- [na] Content Density (Cozy/Compact/(Condensed))
- [na] States - hover/disabled/focused/active/on click/selected/selected hover/press state
- [na] Interaction/Animation - open/close, expand/collapse, add/remove, check/uncheck
- [na] Mouse vs. Keyboard support
- [na] Text Truncation
2. API and functional correctness
- [na] check for console logs (warnings, errors)
- [na] API boundary values
- [na] different combinations of components - free style
- [na] change the API values during testing
3. Documentation and Example validations
- [na] missing API documentation or it is not understandable
- [na] poor examples
- [na] Stackblitz works for all examples
4. Accessibility testing
5. Browser Testing - Edge, Safari, Chrome, Firefox

### CODE REVIEW

File: libs/core/src/lib/calendar/calendar.component.ts [hunk 1]
```diff
@@ -83,13 +84,24 @@ export type DaysOfWeek = 1 | 2 | 3 | 4 | 5 | 6 | 7;
 })
 export class CalendarComponent<D> implements OnInit, ControlValueAccessor, Validator, OnDestroy {
     /** @hidden */
-    @ViewChild(CalendarDayViewComponent) dayViewComponent: CalendarDayViewComponent<D>;
+    @ViewChild(CalendarDayViewComponent)
```
Comment (platon-rov) [inline libs/core/src/lib/calendar/calendar.component.ts:87]: Maybe as these fields are `@hidden` we should prefix their names with `_`?

File: libs/core/src/lib/calendar/calendar-views/calendar-month-view/calendar-month-view.component.ts [hunk 2]
```diff
@@ -121,26 +181,38 @@ export class CalendarMonthViewComponent<D> implements OnInit, OnDestroy, OnChang
     }
 
     /** Method for handling the keyboard events (a11y) */
-    onKeydownMonthHandler(event, index: number): void {
+    onKeydownMonthHandler(event: KeyboardEvent, index: number): void {
```
Comment (platon-rov) [inline libs/core/src/lib/calendar/calendar-views/calendar-month-view/calendar-month-view.component.ts:184]: Maybe this one also should be prefixed with `_`?

**Review by platon-rov — CHANGES_REQUESTED** [review]
Great work!

Please fix Travis.

**Review by InnaAtanasova — APPROVED** [review]

**Review by Smiranin — APPROVED** [review]

**Review by Betrozov — APPROVED** [review]

**Comment by puru-hk** [comment]
Issue 1: Fixed
Issue 2: Fixed
Issue 3: Fixed
Issue 4: Fixed
Issue 5: The dialog is read twice to the user.
Issue 6: Same issue persists
Issue 7: Fixed
Issue 8: The Calender week number is not announced
Issue 9: Same issue persists
Issue 10: Fixed
Issue 11: Same issue persists
Issue 12: Fixed

**Comment by dimamarksman** [comment]
> Issue 1: Fixed
> Issue 2: Fixed
> Issue 3: Fixed
> Issue 4: Fixed
> Issue 5: The dialog is read twice to the user.
> Issue 6: Same issue persists
> Issue 7: Fixed
> Issue 8: The Calender week number is not announced
> Issue 9: Same issue persists
> Issue 10: Fixed
> Issue 11: Same issue persists
> Issue 12: Fixed

@puru-hk thanks for testing.
I've pushed the fix for issue 6 (added title attribute).

Regarding other points that you marked as not fixed. I'm testing it using the vioceover on mac and can not reproduce it.
 * Issue 5. Dialog is read out only once for me.
    
image

* Issue 8
  
image

* Issue 9
  
image
image

* Issue 11
  
image

**Review by mikerodonnell89 — APPROVED** [review]

**Comment by platon-rov** [comment]
@dimamarksman should it be reopened?

**Comment by dimamarksman** [comment]
I need to retest it on windows with JAWS, but currently busy with higher priority task

**Pull request closed without merging**